File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}
    for p in participant:
        dic[p] = dic.get(p, 0) + 1
        
    for c in completion:
        if c in dic and dic[c] > 1:
            dic[c] -= 1
        elif c in dic and dic[c] == 1:
            del dic[c]
            
    return list(dic.keys())[0]
            
            
    
                
            

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const dic = {}
    participant.forEach(v => {
        dic[v] = (dic[v] ?? 0) + 1
    });
    
    completion.forEach(v => {
        dic[v] -= 1;
        if (dic[v] === 0) delete dic[v];
    });
    
    return Object.keys(dic)[0]
}